zerver/tornado/event_queue.py

```diff
@@ -1064,11 +1066,23 @@ def send_notification_http(realm: Realm, data: Mapping[str, Any]) -> None:
     else:
         process_notification(data)
 
-def send_event(realm: Realm, event: Mapping[str, Any],
-               users: Union[Iterable[int], Iterable[Mapping[str, Any]]]) -> None:
+captured_events = {}  # type: Dict[int, Any]
+instrumented_event_counter = 0
+def send_event(realm: Realm, event: Dict[str, Any],
+               users: Union[Iterable[int], Iterable[Mapping[str, Any]]],
+               description: Optional[str]=None) -> None:
     """`users` is a list of user IDs, or in the case of `message` type
     events, a list of dicts describing the users and metadata about
     the user/message pair."""
+    if settings.LOG_API_EVENT_TYPES:
+        global instrumented_event_counter
+        instrumented_event_id = instrumented_event_counter
+        instrumented_event_counter += 1
+        event['instrumented_event_id'] = instrumented_event_id
+
+        copied_event = copy.deepcopy(event)  # type: Dict[str, Any]
+        copied_event['event_description'] = description
```

Comment:
@timabbott  We donesn't seem to be using value of `event_description` anywhere at the moment? Should I make any changes to display it?

Reply:
I added comments explaining these.  The `event_description` field is not used yet but the intent is it'll let us write in-code documentation that'll appear in these API docs.